README.md

```diff
@@ -94,6 +97,183 @@ see [Developer’s guide to auth with Azure Resource Manager API](http://aka.ms/
+
+```ruby
+provider = MsRestAzure::ApplicationTokenProvider.new(
+       ENV['AZURE_TENANT_ID'],
+       ENV['AZURE_CLIENT_ID'],
+       ENV['AZURE_CLIENT_SECRET'])
+credentials = MsRest::TokenCredentials.new(provider)
+
+options = {
+  credentials: credentials,
+  subscription_id: ENV['AZURE_SUBSCRIPTION_ID']
+}
+
+profile_client = Azure::Profiles::V2017_03_09::Mgmt::Client.new(options)
+```
+
+The profile client could be used to access individual RPs:
+
+ ```ruby
+# To access the operations associated with Compute
+profile_client.compute.virtual_machines.get 'RESOURCE_GROUP_NAME', 'VIRTUAL_MACHINE_NAME'
+
+# Option 1: To access the models associated with Compute
+purchase_plan_obj = profile_client.compute.model_classes.purchase_plan.new
+
+# Option 2: To access the models associated with Compute
+purchase_plan_obj = Azure::Profiles::V2017_03_09::Compute::Mgmt::Models::PurchasePlan.new
+
+```
+
+## Usage of Individual gem using Profiles
+
+### Install
+
+You can install the individual gems using gem install. For eg, to install azure_mgmt_compute, use the following command:
+
+```ruby
+gem install 'azure_mgmt_compute'
+```
+### Usage
+
+The following lines should be used to instantiate a profile client:
+
+```ruby
+provider = MsRestAzure::ApplicationTokenProvider.new(
+       ENV['AZURE_TENANT_ID'],
+       ENV['AZURE_CLIENT_ID'],
+       ENV['AZURE_CLIENT_SECRET'])
+credentials = MsRest::TokenCredentials.new(provider)
+
+options = {
+  credentials: credentials,
+  subscription_id: ENV['AZURE_SUBSCRIPTION_ID']
+}
+
+profile_client = Azure::Compute::Profiles::Latest::Mgmt::Client.new(options)
+```
+The profile client could be used to access operations and models:
+
+ ```ruby
+# To access the operations associated with Compute
+profile_client.virtual_machines.get 'RESOURCE_GROUP_NAME', 'VIRTUAL_MACHINE_NAME'
+
+# Option 1: To access the models associated with Compute
+purchase_plan_obj = profile_client.model_classes.purchase_plan.new
+
+# Option 2: To access the models associated with Compute
+purchase_plan_obj = Azure::Compute::Profiles::Latest::Mgmt::Models::PurchasePlan.new
+
+```
+
+## Usage of Individual gem using Versions
+
+In the previous approach, we used the profile associated with individual gem. In the current approach, we could use the
+version directly.
+
+### Install
+
+You can install the individual gems using gem install. For eg, to install azure_mgmt_compute, use the following command:
+
+```ruby
+gem install 'azure_mgmt_compute'
+```
+### Usage
+
+The following lines should be used to instantiate a profile client:
+
+```ruby
+provider = MsRestAzure::ApplicationTokenProvider.new(
+       ENV['AZURE_TENANT_ID'],
+       ENV['AZURE_CLIENT_ID'],
+       ENV['AZURE_CLIENT_SECRET'])
+credentials = MsRest::TokenCredentials.new(provider)
+
+options = {
+  credentials: credentials,
+  subscription_id: ENV['AZURE_SUBSCRIPTION_ID']
+}
+
+compute_client = Azure::Compute::Mgmt::V2016_03_30::ComputeManagementClient.new(credentials)
```

Comment:
add comment "Target specific api-version of Compute, 2016-03-30 in this case"

Reply:
Done